packages/firestore/src/remote/datastore.ts

```diff
@@ -131,7 +125,7 @@ class DatastoreImpl extends Datastore {
       });
   }
 
-  async termiate(): Promise<void> {
+  async terminate(): Promise<void> {
```

Comment:
Unrelated question to this PR, but why are we making `DatastoreImpl.terminate()` async if the method isn't? Why does the base `Datastore` need to have an abstract `terminate()` that returns `Promise<void>`, considering the only implementation of the class is `DatastoreImpl`?

Reply:
I don't have an answer for you, so I changed this as well. In general, returning a Promise even when it is not needed can be seen as "future proofing", but this is only really needed for Public APIs that we don't want to break if there is ever a need to make them async.